wandb/sdk/wandb_config.py

```diff
@@ -152,6 +152,11 @@ def update(self, d, allow_val_change=False):
         if self._callback:
             self._callback(data=self._as_dict())
 
+    def persist(self):
+        """Calls the callback if it's set"""
+        if self._callback:
+            self._callback(data=self._as_dict())
```

Comment:
@raubitsj added `config.persist()`, I'm not sure if this actually persists or if this is the right thing to do.  I know there's code in the wild that calls this method, and I added it because of add_singleton below.

Reply:
Yes, that will persist. 
I dont think it needs to be public though.  Only an issue when changing nested dictionaries? do we want to support auto persisting then (adrian 1 had some code to do that with the summary objects)

Reply:
actually, i guess it is already in our api, and someone might be using it, we can turn it into a noop if we get auto-persisting dictionary updates added.